### DISCUSSION TITLE
Trouble with starting session and passing flags to peaclock

### DISCUSSION THREAD

**Discussion by omnigenous** [kovidgoyal/kitty#7522] [Q&A] [answered]
`kitty --session test.session`

test.session:
```
layout tall
launch peaclock --config ~/.config/peaclock/1 # doesn't work
launch peaclock --config ~/.config/peaclock/2 # doesn't work
launch cmatrix -C red # works
```

cmatrix has no troubles and reads `-C red` flag but peaclock on other hand doesn't. I tested for silly typos, copied `peaclock --config ~/.config/peaclock/1`, pasted to terminal and it did read config with name `1` without an issue.

**Comment by kovidgoyal** [comment]
launch is not going to expand ~ for you. Use the full path. Or if you want to use ~ then do launch sh -c "exec peaclock ~/whatever"

**Reply by omnigenous** [reply]
Changed to `launch sh -c "exec peaclock --config ~/.config/peaclock/1"` still doesn't work. Changing to full path also doesn't change anything `launch sh -c "exec peaclock --config /home/user/.config/peaclock/1"`.

**Reply by kovidgoyal** [reply]
Then I have no idea.

**Comment by omnigenous** [comment, marked as answer]
I had an alias `alias peaclock=peaclock --config-dir ~/.config/peaclock`, didn't thought it would matter when providing `--config` flag but it is.

```
layout tall
launch sh -c "exec peaclock --config-dir ~/.config/peaclock --config ~/.config/peaclock/1" # works now
launch sh -c "exec peaclock --config-dir ~/.config/peaclock --config ~/.config/peaclock/2" # works now
launch cmatrix -C red # works
```
